### DISCUSSION TITLE
Export ADIF with QO-100 QSOs for LoTW

### DISCUSSION THREAD

**Discussion by dl1com** [magicbug/Cloudlog#1093] [Q&A]
Hi,
first of all: thanks to @magicbug and all the contributors for Cloudlog, I really love it!

Now, I want to export/import my QSOs to/from LoTW and run into several issues:

0. Where can I find the Cloudlog version I am running?
I downloaded the current state of the master branch a few weeks ago. Is there a place on the Webinterface where I can see the commit hash I am running on?

1. Manual sync with LoTW throws an error
  I uploaded my certificate, which worked fine. Now I wanted to sync with LoTW, but Cloudlog shows this error.
  I checked the credentials multiple times. Folders have also the correct permissions (according to Admin -> Debug Information).
  The issue #331 suggests that this error is caused by an empty LoTW log (which is the case for me). Can you confirm this?

  ```
  LoTW Matches
  A PHP Error was encountered
  Severity: Notice
  
  Message: Trying to get property 'COL_LOTW_QSLRDATE' of non-object
  
  Filename: models/Logbook_model.php
  
  Line Number: 1409
  
  LoTW Downloading failed either due to it being down or incorrect logins.
  ```

2. SAT_NAME not set in ADIF Export
  As the manual LoTW sync in Cloudlog does not work for me, I tried to export ADIF from Cloudlog and upload it using TQSL.
  When trying to sign and upload it to LoTW, it tells me that the "SAT_NAME" field is not set. And really, there is no SAT_NAME 
  field in the exported ADIF file, although "Sat Name" is set in the QSO entries.
  How can I trigger to have this field exported to ADIF?

3. DXCC Entity mismatch
  I tried to upload some other (non-sat) QSOs to verify...but I also failed with them, as TQSL tells me that the DXCC I chose for LoTW (Federal Republic of Germany) does not match with the Country in the QSOs (just Germany). As LoTW only accepts Federal Republic of Germany and not Germany when setting up a callsign: How can I fix my QSOs to have the correct DXCC?
  The cloudlog webinterfac shows me "Federal Republic Of Germany" for these QSOs, but obviously instead of the DXCC code 230 which LoTW expects it exports 81, which seems to be obsolete.

Thanks a lot for your help, and sorry in advance in case I missed something in the documentation!

**Comment by dl1com** [comment]
Hi,
in the meantime, I managed to solve points 1. and 3. of my original post:
The DXCC can be set up in the station location menu of cloudlog. By intuition, I chose "Germany" over "Federal Republic of Germany", which does not work with LoTW. After setting the station location to "Federal ...", I was finally able to do a manual sync with LoTW!
During the first two manual syncs, it still showed the `Trying to get property 'COL_LOTW_QSLRDATE' of non-object` error, but then it was gone.

So I would propose to mark obsolete DXCCs in the station location menu as such (but still provide them), so one does not accidentially use them.

Points 0. and 2. are still open, and I'd love to get some answer/opinion on them :)

**Comment by AndreasK79** [comment]
Hi @chris007de. 

0. Identifying version is tricky, as version numbering is not upped very often. The only way to know, is to look for functions from the latests commits.
2. I had a look at the code, and the LoTW export does export the satname:
`<?php if($qso->COL_SAT_NAME) { 
	$satellite_name_check = $CI->lotw_satellite_map(strtoupper($qso->COL_SAT_NAME)); 
	if($satellite_name_check != FALSE) { 
		$satname = $satellite_name_check; 
	} else { 
		$satname = $qso->COL_SAT_NAME; 
	} ?>
<SAT_NAME:<?php echo strlen($satname); ?>><?php echo strtoupper($satname); ?><?php 
} ?>`
 
The regular adif export does this if COL_SAT_MODE is set:
`     if ($qso->COL_SAT_NAME) {
            if ($qso->COL_SAT_MODE != 0 || $qso->COL_SAT_MODE !="") {
                $line .= $this->getAdifFieldLine("SAT_MODE", $qso->COL_SAT_MODE);
                $line .= $this->getAdifFieldLine("SAT_NAME", $qso->COL_SAT_NAME);
            }
        }`

Hope this clears up your questions.
